core/src/utils/periodic_task_test.cpp

```diff
@@ -418,4 +418,39 @@ UTEST_F(PeriodicTaskLog, ErrorLog) {
     task.Stop();
 }
 
+UTEST(PeriodicTask, EnableDisable) {
```

Comment:
Add a test that checks that if `enabled=false` during `Start` (even if `enabled=true` in `PeriodicTask` constructor), then the first iteration won't happen immediately and won't happen after `period`